packages/next/src/client/components/params.ts

```diff
@@ -1 +1,88 @@
+import { trackDynamicDataAccessed } from '../../server/app-render/dynamic-rendering'
+import { ReflectAdapter } from '../../server/web/spec-extension/adapters/reflect'
+import { getRouteMatcher } from '../../shared/lib/router/utils/route-matcher'
+import { getRouteRegex } from '../../shared/lib/router/utils/route-regex'
+import { staticGenerationAsyncStorage } from './static-generation-async-storage.external'
+
+export type UnknownRouteParams =
+  | ReadonlySet<string>
+  | ReadonlyMap<string, string | string[]>
+  | undefined
```

Comment:
consider

```
type UnknownFallbackRouteParams = ReadonlySet<string>
type KnownFallbackRouteParams = ReadonlyMap<string, string | string[]>
type FallbackRouteParams = KnownFallbackRouteParams | UnknownFallbackRouteParams

{
  fallbackParams: null | FallbackRouteParams
}
```

This way you can define functions which assert you've already ensured we have one of the two types (the null is external) and it avoids the confusion of unknown...AreKnown or KnownUnknown...